code/datums/supply_packs/black_market.dm

```diff
@@ -0,0 +1,1308 @@
+		// That was the good 50%. Now it's time for the bad.
+		if(51 to 55)
+		// Random junk
+			new /obj/effect/essentials_set/random/clf_bonus_item(loc)
+			new /obj/effect/essentials_set/random/clf_bonus_item(loc)
+			new /obj/effect/essentials_set/random/clf_bonus_item(loc)
+			new /obj/effect/essentials_set/random/clf_bonus_item(loc)
+			new /obj/effect/essentials_set/random/clf_bonus_item(loc)
+			loot_message = SPAN_NOTICE("It's a bunch of random junk...")
+		if(56 to 60)
+		// Egg cluster, oh no!
+			new /obj/effect/spider/eggcluster(loc)
+			loot_message = SPAN_WARNING("What the hell is this?")
+		if(61 to 65)
+		// Actual spider!!
+			new /mob/living/simple_animal/hostile/giant_spider(loc)
+			loot_message = SPAN_DANGER("It's a giant spider!!")
+		if(66 to 70)
+		// Actual dead spider!!
+			var/mob/living/spider = new /mob/living/simple_animal/hostile/giant_spider(loc)
+			spider.death()
+			loot_message = SPAN_DANGER("It's a giant spider!! Oh, guess there were no air holes.")
+		if(71 to 75)
+		// Many small spiders. Yes, there's a 20% chance of something spider-related coming out of this box.
+			for(var/i in 1 to 10)
+				new /obj/effect/spider/spiderling(loc)
+			loot_message = SPAN_DANGER("SPIDERS!! AAAAAAAH!!")
+		if(76 to 80)
+		// CLF corpse!! Why is this here? Don't ask.
+			var/mob/living/carbon/human/corpse = new /mob/living/carbon/human(loc)
+			corpse.create_hud() //Need to generate hud before we can equip anything apparently...
+
+			var/corpse_type = pick(/datum/equipment_preset/corpse/clf/burst, /datum/equipment_preset/corpse/clf)
+			arm_equipment(corpse, corpse_type, TRUE, FALSE) // I didn't choose the shitcode life, the shitcode life chose me
+
+			loot_message = SPAN_HIGHDANGER("IT'S A CORPSE!!")
+		if(81 to 85)
+		// Costumes.
+			new /obj/effect/landmark/costume/random(loc)
+			new /obj/effect/landmark/costume/random(loc)
+			new /obj/effect/landmark/costume/random(loc)
+			loot_message = SPAN_NOTICE("What the hell is this..?")
+		if(86 to 90)
+		// Random supply garbage.
+			new /obj/effect/spawner/random/tool(loc)
+			new /obj/effect/spawner/random/tool(loc)
+			new /obj/effect/spawner/random/tech_supply(loc)
+			new /obj/effect/spawner/random/tech_supply(loc)
+			new /obj/effect/spawner/random/tech_supply(loc)
+			new /obj/effect/spawner/random/toy(loc)
+			new /obj/effect/spawner/random/toy(loc)
+			new /obj/effect/spawner/random/toy(loc)
+			loot_message = SPAN_NOTICE("It's just a bunch of junk!")
+		if(91 to 95)
+		// We don't really have any other kind of booby trap so this will do
+			new /obj/item/explosive/grenade/spawnergrenade/claymore_launcher(loc)
+			loot_message = SPAN_HIGHDANGER("It was booby trapped! RUN!")
+		if(96 to 99)
+		// Oh boy. Big booby trap!
+			var/obj/item/mortar_shell/frag/fragshell = new(loc)
+			var/obj/item/explosive/grenade/incendiary/molotov/molotov = new(loc)
+			molotov.prime()
+			fragshell.balloon_alert_to_viewers("the mortar shell makes an awful hissing noise!")
```

Comment:
```suggestion
			fragshell.balloon_alert_to_viewers("[fragshell] makes an awful hissing noise!")
```

Reply:
![image](https://user-images.githubusercontent.com/53100513/215288632-43e0d3dc-8b5a-4e26-8cc4-14ece827c957.png)
normally i'd agree, but this isn't visible_message, it's balloon_alert, and making it the full name results in massive screen bloat. i know from experience trying it